src/dialogsearch.cpp

```diff
@@ -65,7 +65,32 @@ void DialogSearch::on_button_search_clicked()
     sl << "Map" << "Event" << "Event Page" << "Sourceline" << "Action";
     ui->list_result->setHorizontalHeaderLabels(sl);
 
-    std::function<std::vector<command_info>(const RPG::Map&)> map_searcher;
+    std::function<bool(const RPG::EventCommand&)> search_predicate;
+    auto map_searcher = [&search_predicate](const RPG::Map& map) {
+        std::vector<command_info> res;
+        for (auto& e : map.events)
+            for (auto& p : e.pages)
+                for (size_t line = 0; line < p.event_commands.size(); line++)
+                {
+                    auto& com = p.event_commands[line];
+                    if (search_predicate(com))
+                        res.emplace_back(map.ID, e.ID, p.ID, line, com);
```

Comment:
Internally the Line numbers starts from 0 in RPG_RT but when this is exposed to the user I would prefer a numbering from 1. The typical RPG Maker user is usually not a programmer and this will lead to confusion (everything else is already 1-based in the Maker Editor)